mod_modular_vanilla/hooks/skills/skill_container.nut

```diff
@@ -0,0 +1,47 @@
+// MV: Added
+// Part of the actor.interrupt framework
+::MSU.Skills.addEvent("onActorInterrupted", function( _offensive, _defensive ) {});
+
+::ModularVanilla.QueueBucket.VeryLate.push(function() {
+	::ModularVanilla.MH.hook("scripts/skills/skill_container", function(q) {
+		q.m.MV_InterruptionFrame <- 0; // Part of the actor.interrupt framework
+		q.m.MV_InterruptionCount <- 0; // Part of the actor.interrupt framework
+
+		// Part of actor.interrupt framework
+		// In vanilla skills which "interrupt" a character remove: effects.shieldwall, effects.spearwall, effects.riposte
+		// immediately one after the other i.e. it happens within a single frame. So we check if all 3 effects were
+		// removed in a single frame and assume that the vanilla intention is to trigger an interruption of the actor.
+		// This is done instead of hooking every single vanilla file which triggers these kinds of interruptions.
+		// This implementation should also cover all mods which followed the vanilla style of removing those 3 effects only.
+		// An exception is the disarmed_effect which doesn't remove shieldwall. For that we hook that effect directly and do
+		// offensive interrupt only.
+		q.removeByID = @(__original) function( _skillID )
+		{
+			switch (_skillID)
+			{
+				case "effects.shieldwall":
+				case "effects.spearwall":
+				case "effects.riposte":
+					local frame = ::Time.getFrame();
```

Comment:
How does `getFrame()` work exactly? What does it do?
I don't have any experience with that function.
I see it being used a few times in vanilla and being implemented in the .exe.

Reply:
This is also my first time working with this function. In my testing in-game it seems to me that it prints the total number of "frames" of the game that have run since the start of the game. And I assume that a given sequence of code and function calls in squirrel runs inside a single frame.